Modules/Core/Common/include/itkProcessObject.h

```diff
@@ -297,25 +297,35 @@ class ITKCommon_EXPORT ProcessObject : public Object
   /** \brief Turn on and off the AbortGenerateData flag. */
   itkBooleanMacro(AbortGenerateData);
 
-  /** \deprecated
-   * Set the execution progress of a process object. The progress is
-   * a floating number in [0,1] with 0 meaning no progress and 1 meaning
-   * the filter has completed execution.  The ProgressEvent is NOT
-   * invoked.
-   * This method is deprecated because filters should not be calling
-   * SetProgress directly but should be using UpdateProgress instead.
-   * We avoid the use of the itkSetClampMacro because that macro calls
-   * Modified on the filter, which will cause the filter to rerun even
-   * if it doesn't need to.
-   * Thus, we implement the SetClampMacro directly without the call to
-   * Modified. */
+  static constexpr float
+  progressFixedToFloat(uint32_t f)
```

Comment:
Why 32-bit integer? We already commonly process images bigger than that. If this is meant to be performant on 32-bit processors, we could use `SizeValueType`. Answering self: 32-bit float has less significant bits and 32-bit integer, so it is a fine replacement.